src/main/java/org/liliya/hotelapp/service/ApartmentServiceImpl.java

```diff
@@ -1,92 +1,202 @@
 package org.liliya.hotelapp.service;
 
-import com.fasterxml.jackson.core.type.TypeReference;
 import org.liliya.hotelapp.configuration.Configuration;
 import org.liliya.hotelapp.model.Apartment;
 import org.liliya.hotelapp.model.Client;
 import org.liliya.hotelapp.model.ReservationStatus;
-import org.liliya.hotelapp.persistence.StatePersistence;
+import org.liliya.hotelapp.persistence.DatabaseConnection;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 import java.util.ArrayList;
-import java.util.Comparator;
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class ApartmentServiceImpl implements ApartmentService {
-    private final List<Apartment> apartments = new ArrayList<>();
-    private int idCounter = 1;
     private final Configuration configuration;
-    private final StatePersistence<List<Apartment>> statePersistence;
+    private final DatabaseConnection databaseConnection;
 
-    public ApartmentServiceImpl(StatePersistence<List<Apartment>> statePersistence, Configuration configuration) {
-        this.configuration = configuration;
-        this.statePersistence = statePersistence;
-        addAllApartments();
+    public ApartmentServiceImpl(DatabaseConnection databaseConnection) {
+        this.configuration = Configuration.getInstance();
+        this.databaseConnection = databaseConnection;
     }
 
     @Override
-    public void register(double price) {
-        Apartment apartment = new Apartment(idCounter++, price, ReservationStatus.AVAILABLE, null);
-        apartments.add(apartment);
+    public int register(double price) {
+        String query = "INSERT INTO apartments (price, status) VALUES (?, ?)";
+        int generatedId = -1;
+        try (Connection connection = databaseConnection.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS)
+        ) {
+            statement.setDouble(1, price);
+            statement.setString(2, ReservationStatus.AVAILABLE.name());
+
+            int rowsInserted = statement.executeUpdate();
+            if (rowsInserted > 0) {
+                try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                    if (generatedKeys.next()) {
+                        generatedId = generatedKeys.getInt(1);
+                        System.out.println("Apartment registered with ID: " + generatedId);
+                    }
+                }
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return generatedId;
     }
 
     @Override
     public boolean reserve(Client client) {
```

Comment:
in general split method with small request.
1) insertClient
2)findAvailable
3) updateApartmen
and use just here))